drivers/timer/stm32_lptim_timer.c

```diff
@@ -43,6 +43,10 @@
  * This is for example about of 65000 x 2000ms when clocked by LSI
  */
 static uint32_t accumulated_lptim_cnt;
+/* Next autoreload value to set */
+static uint32_t autoreload_next;
+/* Indicate if the autoreload register is ready for a write */
+static bool autoreload_ready;
```

Comment:
why could the bool autoreload_ready not be  LL_LPTIM_ClearFlag_ARROK ?

Reply:
In interrupt, when the flag ARROK is cleared and the autoreload register is not set (because the register is already at the correct value), the information that the autoreload register is ready is lost. It's why I use a variable to store this information and use it in sys_clock_set_timeout.

Reply:
Also to be init'd `autoreload_ready = 1;`

Reply:
The autoreload register is set during the init function (sys_clock_driver_init), it's why I set autoreload_ready to 0 at startup.

Reply:
Ok, 
 following the code execution, AutoReload reg is first set during `sys_clock_driver_init()`, then the ARROK interrupt is triggered. There `autoreload_next` seems not to be set yet but autoreload_next == autoreload so the _if_   at line 63 goes to   `autoreload_ready = 1`

Reply:
I made a function to update autoreload_next and  autoreload_ready. I think this can remove some ambiguities